src/UsersEpic.js

```diff
@@ -0,0 +1,43 @@
+//@flow
+import {ofType} from 'redux-observable';
+import {ACTIONS} from 'redux-api-call';
+import {map, filter} from 'rxjs/operators';
+import {addUsers} from './UsersAction';
+import {combineEpics} from 'redux-observable';
+import {fetchUsersAC} from './state.js';
+
+const fetchUsersCompleteEpic = action$ =>
+  action$.pipe(
+    ofType(ACTIONS.COMPLETE),
+    filter(action => action.payload.name === 'FETCH_USERS'),
+    map(({payload}) => {
+      const fetchedUsers = payload.json.results || [];
+      return addUsers(fetchedUsers);
+    }),
+  );
+
+let fetchUsersNetworkFails: number = 0;
+const fetchUsersNetworkFailureEpic = action$ =>
+  action$.pipe(
+    ofType(ACTIONS.FAILURE),
+    filter(
+      ({payload}) =>
+        payload.name === 'FETCH_USERS' &&
+        payload.json.message === 'Network request failed',
+    ),
+    map(res => {
+      if (++fetchUsersNetworkFails < 3) {
+        return fetchUsersAC();
+      } else {
+        fetchUsersNetworkFails = 0;
+        return {type: 'NO_ACTION'};
```

Comment:
Suppose a case that
- App sends request #1, the API got failed => app will retry with request #2
- Request #2 is successful but fetchUsersNetworkFails is not reset value

Reply:
Thank you for your feedback. I definitely missed that case here. I've added the code to reset fetchUsersNetworkFails when a request completes.